### DISCUSSION TITLE
Correct rules regarding document URL's for subtypes 4, 5, 6?

### DISCUSSION THREAD

**Discussion by Colan71** [OP-TED/eForms-SDK#766] [Q&A] [answered]
Dear Team,
Are the conditionally mandatory rules for BT-15 Documents URL and BT-615 Documents restricted URL really applicable to subtypes 4, 5 and 6 considering that they are notice types Prior information?

  **"id" : "BT-15-Part",**
    "parentNodeId" : "ND-PartProcurementDocument",
    "name" : "Documents URL",
    "btId" : "BT-15",
    "xpathAbsolute" : "/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Part']/cac:TenderingTerms/cac:CallForTendersDocumentReference/cac:Attachment[../cbc:DocumentType/text()='non-restricted-document']/cac:ExternalReference/cbc:URI",
    "xpathRelative" : "cac:Attachment[../cbc:DocumentType/text()='non-restricted-document']/cac:ExternalReference/cbc:URI",
    "xsdSequenceOrder" : [ { "cac:Attachment" : 16 }, { "cac:ExternalReference" : 4 }, { "cbc:URI" : 2 } ],
    "type" : "url",
    "legalType" : "URL",
    "maxLength" : 400,
    "repeatable" : {
      "value" : false,
      "severity" : "ERROR"
    },
    "forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "1", "2", "3", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "37", "38", "39", "40", "CEI", "T01", "T02", "X01", "X02" ],
        "value" : true,
        "severity" : "ERROR"
      }, {
        **"noticeTypes" : [ "4", "5", "6" ],
        "condition" : "{ND-PartProcurementDocument} ${not(BT-14-Part == 'non-restricted-document')}",
        "value" : true,
        "severity" : "ERROR"
      } ]
    },
    "mandatory" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "4", "5", "6" ],
        "value" : true,
        "severity" : "ERROR"**
      } ]
    },
    "pattern" : {
      "value" : "((^(http|HTTP|https|HTTPS|ftp|FTP|ftps|FTPS|sftp|SFTP)://)|(^(w|W){3}(\\d)?\\.))[\\w\\?!\\./:;,\\-_=#+*%@&quot;\\(\\)&amp;]+",
      "severity" : "ERROR"
    }

 **"id" : "BT-615-Part",**
    "parentNodeId" : "ND-PartProcurementDocument",
    "name" : "Documents Restricted URL",
    "btId" : "BT-615",
    "xpathAbsolute" : "/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Part']/cac:TenderingTerms/cac:CallForTendersDocumentReference/cac:Attachment[../cbc:DocumentType/text()='restricted-document']/cac:ExternalReference/cbc:URI",
    "xpathRelative" : "cac:Attachment[../cbc:DocumentType/text()='restricted-document']/cac:ExternalReference/cbc:URI",
    "xsdSequenceOrder" : [ { "cac:Attachment" : 16 }, { "cac:ExternalReference" : 4 }, { "cbc:URI" : 2 } ],
    "type" : "url",
    "legalType" : "URL",
    "maxLength" : 400,
    "repeatable" : {
      "value" : false,
      "severity" : "ERROR"
    },
    "forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "1", "2", "3", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "37", "38", "39", "40", "CEI", "T01", "T02", "X01", "X02" ],
        "value" : true,
        "severity" : "ERROR"
      }, {
        **"noticeTypes" : [ "4", "5", "6" ],
        "condition" : "{ND-PartProcurementDocument} ${not(BT-14-Part == 'restricted-document')}",**
        "value" : true,
        "severity" : "ERROR"
      } ]
    },
    **"mandatory" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "4", "5", "6" ],
        "value" : true,
        "severity" : "ERROR"**
      } ]
    },
    "pattern" : {
      "value" : "((^(http|HTTP|https|HTTPS|ftp|FTP|ftps|FTPS|sftp|SFTP)://)|(^(w|W){3}(\\d)?\\.))[\\w\\?!\\./:;,\\-_=#+*%@&quot;\\(\\)&amp;]+",
      "severity" : "ERROR"

**Comment by YvesJo** [comment, marked as answer]
Hi,
As soon as you refer to procurement documents (BG-701), then you have to provide the URLs. Given the context node and the rule following which no empty node or element should be present in the XML, if you encounter an issue with that rule, then most likely the generated XML is not conform. The presence of the other BG-701 Business Terms without a URL to the documents wouldn't be very useful. When a PIN Only is considered too early for your procedures, then do not provide any information on procurement documents, whatever these are.
KR,

**Comment by Colan71** [comment]
Hi,
Not sure I quite get it. Does this mean that we can send PIN only notices (subtypes 4, 5, 6) without URLs and they will not get stuck in validation even though the rules states they are mandatory?

**Reply by YvesJo** [reply]
What is meant, is that in the case you use any information related to procurement documents (cf. https://docs.ted.europa.eu/eforms/latest/schema/procedure-lot-part-information.html#procurementDocsSection) then you'll have to specify the appropriate URL and this independently from the fact that this is a PIN Only notice or not. If you do not wish to include information of procurement documents in your PIN Only notices, feel free to do so, just make sure you do not use any of the associated fields (including the document identifier OPT-140).
